### DISCUSSION TITLE
Github Copilot Conflicts with vscode user snippets

### DISCUSSION THREAD

**Discussion by Ptolemy2002** [community/community#7012] [Copilot Conversations]
**description**:

When you initiate a user snippet in vscode and GitHub Copilot has a suggestion, hitting the tab button will accept the suggestion, but hitting the escape button will cancel your snippet entirely.

Additionally, if you do accept the suggestion, it doesn't seem to register as something you actually wrote, which can be problematic for snippets with regex replacements.

**steps to reproduce**:

1. Open a folder in vscode
2. Go to File > Preferences > User Snippets > and choose "javascript.json"
2. Append [this user snippet](https://pastebin.com/L7RsBEbm) to the end of the JSON Object and save your changes.
3. Create a file called "test.js" and open it.
4. Type "test" and let vscode autocomplete fill in your snippet.
5. Begin to type "convert a string to a number". Github Copilot should suggest the full description.
6. Press the ESC button. The suggestion is not dismissed, but the user snippet is canceled.
7. Accept the suggestion and move your cursor to the part after "return type."
8. Begin to type "integet". Github Copilot should suggest it in full.
9. Accept the suggestion. The text is not replaced with "Number" even though our snippet is defined to do so.

**Expected Behavior**:

On step 6, the suggestion is dismissed without canceling the snippet.
On step 9, the text "integet" should be replaced with "Number".

Attached is a video of this bug being reproduced followed by an example of how to "get around" it, producing more expected behavior

https://user-images.githubusercontent.com/32689317/139458072-961fe56f-f993-4cad-9691-fa471fdd9dfe.mp4

**Comment by Sti2nd** [comment]
A quick solution would be to remap the `accept` keyboard shortcut to <kbd>Shift</kbd> + <kbd>Tab</kbd> for example. See https://github.com/github/copilot-docs/blob/main/docs/visualstudiocode/configuring.md

Though I agree that it would be great if Copilot and VS Code snippets could use the same keybinding, so I created a new feature request https://github.com/github/feedback/discussions/8752

**Comment by bjrmatos** [comment]
just for the record, another quick solution that does not involve remap to a different key is to change just the `when` expression of the `inlineSuggest.commit` to `inlineSuggestionHasIndentationLessThanTabSize && inlineSuggestionVisible && !editorTabMovesFocus && !inSnippetMode`. the addition is `&& !inSnippetMode` which basically prevents the `Tab` to commit the inline suggestion if we are in snippet mode.

<img width="1724" alt="Captura de pantalla 2022-01-07 a la(s) 10 12 51 a  m" src="https://user-images.githubusercontent.com/4262050/148564151-59aef660-e69b-4d4e-b48e-5a5297299a27.png">

**Reply by tushar-appsdeployer** [reply]
thanku for solution

**Comment by nermin99** [comment]
Even though @bjrmatos's solution is a good workaround, I would like to have the option to not show any Copilot suggestion at all while I'm in a snippet.

**Reply by Sti2nd** [reply]
https://github.com/orgs/community/discussions/8752
